### PR TITLE
[Fix] fix CosineRestart eta_min

### PR DESCRIPTION
**Pull request by Z-Fran**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by HAOCHENYE — DISMISSED** [review]

**Comment by RangiLyu** [comment]
add some unit tests to cover this case

File: tests/test_optim/test_scheduler/test_param_scheduler.py [hunk 1]
```diff
@@ -438,6 +438,24 @@ def test_cosine_restart_scheduler(self):
             eta_min=0)
         self._test_scheduler_value(scheduler, targets, epochs=10)
 
+        epochs = 10
+        t = 10
+        eta_min = 1e-7
```
Comment (HAOCHENYE) [inline tests/test_optim/test_scheduler/test_param_scheduler.py:443]: The absolute tolerance given in `_test_scheduler_value` is `1e-5`, `1e-7` is too small to be recognized as an error.

File: tests/test_optim/test_scheduler/test_param_scheduler.py [hunk 2]
```diff
@@ -438,6 +438,24 @@ def test_cosine_restart_scheduler(self):
...
+            eta_min + (0.05 - eta_min) * (1 + math.cos(math.pi * x / t)) / 2
+            for x in range(epochs)
+        ]
+        targets = [
+            single_targets, [x * self.layer2_mult for x in single_targets]
+        ]
+        scheduler = CosineRestartParamScheduler(
+            self.optimizer,
+            param_name='lr',
+            periods=[t],
```
Comment (HAOCHENYE) [inline tests/test_optim/test_scheduler/test_param_scheduler.py:454]: The last epoch will not be tested in `_test_scheduler_value`, therefore `0` learning rate will not be compared with `1e-7` at the tenth epoch.

**Review by HAOCHENYE — COMMENTED** [review]
We should make sure the added unit test will throw an error if the code has not been modified.

**Review by RangiLyu — APPROVED** [review]

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**